text/profiles/0239-profiles-data-model.md

```diff
@@ -0,0 +1,1601 @@
+
+### Design Goals
+
+These goals are based on the vision set out in [Profiling Vision OTEP](./0212-profiling-vision.md):
+
+* Make profiling compatible with other signals
+* Standardize profiling data model for industry-wide sharing and reuse
+* Profilers must be able to be implementable with low overhead and conforming to OpenTelemetry-wide runtime overhead / intrusiveness and wire data size requirements.
+
+The last point is particularly important in the context of profiling. Profilers generate large amounts of data, and users of profiling technology are very sensitive to the overhead that profiling introduces. In the past high overhead has been a blocker for wider adoption of continuous profiling and was one of the reasons why profiling was not used in production environments. Therefore it is important to make sure that the overhead of handling the profiling data on the client side as well as in intermediaries (e.g collector) is minimal.
+
+## Data Model
+
+This section describes various protobuf messages that are used to represent profiles data.
+
+### Relationships Diagram
+
+The following diagram shows the relationships between the messages. Relationships between messages are represented by either embedding one message in another (red arrows), or by referencing a message by index in a lookup table (blue arrows). More on that in [Relationships Between Messages](#relationships-between-messages) section below.
+
+In addition to that, relationship between `samples` and `locations` is further optimized for better performance. More on that in [Relationship Between Samples and Locations](#relationship-between-samples-and-locations) section below.
+
+![diagram of data relationships](./images/otep0239/profiles-data-model.png)
+
+### Relationships With Other Signals
+
+There are two types of relationships between profiles and other signals:
+
+* from other signals to profiles (e.g from log records, exemplars or trace spans)
+* from profiles to other signals
+
+#### From Profiles to Other Signals
+
+[Link](#message-link) is a message that is used to represent connections between profile [Samples](#message-sample) and trace spans. It uses `trace_id` and `span_id` as identifiers.
+
+For other signals, such as logs or metrics, because other signals use the same way of linking between such signals and traces (`trace_id` and `span_id`), it is possible to correlate profiles with other signals using this same information.
+
+#### From Other Signals to Profiles
+
+Other signals can use `profile_id` to reference a profile. For example, a log record can reference a profile that was collected at the time when the log record was generated by using `profile_id` as one of the attributes. This allows to correlate logs with profiles.
+
+Additionally, `trace_id`, `span_id` can be used to reference specific [Samples](#message-sample) in a Profile, since [Samples](#message-sample) are linked to traces with these same identifiers using [Links](#message-link).
```

Comment:
I think this can be slightly misleading since this might create an impression that this allows to link to a given sample, but this is not so because it's really more about linking to a given span and thus also, as a by-product, linking to all samples linked to that span.

Reply:
Addressed it in [this commit](https://github.com/open-telemetry/oteps/pull/239/commits/e61721cda19dab8a7a7db0520540747438e11248)

This is a tricky one, let me know if you think it's still misleading.